pkg/engine/update_test.go

```diff
@@ -49,3 +54,42 @@ func TestAbbreviateFilePath(t *testing.T) {
 		assert.Equal(t, filepath.ToSlash(tt.expected), filepath.ToSlash(actual))
 	}
 }
+
+func TestDeletingCustomResourceProducesResourceOutputsEvent(t *testing.T) {
```

Comment:
Are we testing custom resources or component (non-custom) resources? I _think_ the intention is to actually test component resources, but am I misunderstanding the test?

```suggestion
func TestDeletingComponentResourceProducesResourceOutputsEvent(t *testing.T) {
```